src/composables/transactionList.ts

```diff
@@ -129,160 +128,52 @@ export function useTransactionList({ store }: IDefaultComposableOptions) {
-      return [];
-    }
-  }
-
-  async function fetchPendingTransactions(address: Encoded.AccountAddress) {
-    const sdk = await getAeSdk();
-
-    try {
-      const fetchedPendingTransaction = (
-        await sdk.api.getPendingAccountTransactionsByPubkey(address)
-      );
-      const transactionState = getAccountTransactionsState(address);
-
-      return JsonBig.parse(JsonBig.stringify(
-        fetchedPendingTransaction?.transactions || [],
-      ))
-        .filter(
-          (transaction: ITransaction) => !transactionState.pending[nodeNetworkId.value!]?.find(
-            (tx) => tx?.hash === transaction?.hash,
-          ),
-        ).map((transaction: ITransaction) => ({ ...transaction, pending: true }));
-    } catch (error) {
-      return [];
-    }
-  }
-
-  async function fetchTipWithdrawnTransactions(address: Encoded.AccountAddress, recent: boolean) {
-    try {
-      await getAeSdk();
-      const response = await fetchJson(
-        `${aeActiveNetworkSettings.value.backendUrl}/cache/events/?address=${address}&event=TipWithdrawn${recent ? '&limit=5' : ''}`,
-      );
-      if (response.message) {
-        return [];
-      }
-      // TODO prepare interface for response
-      const tipWithdrawnTransactions: ITransaction[] = (response as any[]).map(({
-        amount,
-        contract,
-        height,
-        data: { tx },
-        ...t
-      }) => ({
-        tx: {
-          ...tx,
-          address,
-          amount,
-          contractId: contract,
-          type: Tag[Tag.ContractCallTx],
-        },
-        ...t,
-        microTime: new Date(t.createdAt).getTime(),
-        blockHeight: height,
-        claim: true,
-      }));
-
-      return tipWithdrawnTransactions;
-    } catch (error) {
-      return [];
-    }
-  }
-
   async function fetchTransactions(
     limit: number,
     recent: boolean,
     address: Encoded.AccountAddress,
+    protocol?: string,
   ): Promise<ITransaction[]> {
-    await getAeSdk(); // Ensure the `nodeNetworkId` is established
-
     ensureAccountTransactionStateExists(address);
+    let preparedTransactions: ITransaction[] = [];
     const transactionState = transactions.value[address];
 
-    if (transactionState.nextPageUrl === null && !recent) {
-      return [];
-    }
-
-    const [
-      regularTransactions,
-      pendingTransactions,
-      tipWithdrawnTransactions,
-    ] = await Promise.all([
-      fetchTransactionsFromMiddleware(transactionState, address, recent, limit),
-      fetchPendingTransactions(address),
-      fetchTipWithdrawnTransactions(address, recent),
-    ]);
-
-    const lastPendingTransaction = pendingTransactions?.[pendingTransactions.length - 1];
-    if (lastPendingTransaction?.type === AEX9_TRANSFER_EVENT) {
-      const middleware = await getMiddleware();
-      pendingTransactions[pendingTransactions.length - 1] = (
-        await middleware.getTx(lastPendingTransaction.payload.txHash)
+    if (protocol === PROTOCOL_AETERNITY) {
```

Comment:
As mentioned before - we should not have any protocol specific logic in the global scope.